staging/src/k8s.io/component-base/logs/registry.go

```diff
@@ -80,6 +80,17 @@ func (lfr *LogFormatRegistry) Delete(name string) {
 	delete(lfr.registry, name)
 }
 
+// List names of registered log formats
+func (lfr *LogFormatRegistry) List() []string {
+	lfr.mu.Lock()
```

Comment:
Why is this lock needed? If a format is registered after the help message is generated, it won't be included. It seems like at some point in time the format list should be frozen and we actually want to panic or error if an errant programmer registers a new one.

(Not asking for a change in this PR, just noting that this is suspicious)

Reply:
Register was preemptively made thread safe, but after feedback from @cheftako I think we will remove lock.
I addes locking to List to be consistent with other methods. I will create separate PR to remove locking